src/utils/body.ts

```diff
@@ -1,6 +1,6 @@
 import { HonoRequest } from '../request'
 
-export type BodyData = Record<string, string | File | (string | File)[]>
+export type BodyData = Record<string, string | object | File | (string | File)[]>
```

Comment:
In place of an `object` type, we should now have a nested generic type that would be more helpful

Reply:
I'm not sure we really need generic type for an empty object `{}` used on dot notation logic, maybe you consider more on the user usage ?

Reply:
I believe this will provide a better type completion compared to having just an object type

```ts
type BodyDataValue = string, string | File | (string | File)[]
export type BodyData = Record<string, BodyDataValue | Record<string, BodyDataValue>>
```

Reply:
Looks like can't use `type` circular references

Reply:
I ended up with this
```js
type BodyDataValue = string | {} | File | (string | File)[] | { [key: string]: BodyDataValue }
export type BodyData = Record<string, BodyDataValue>
```

Looks neat!

Reply:
we can change the linting rules and all. @yusukebe should we change them? or maybe we can use `// @ts-ignore` or something?

Reply:
Reopen it, whether continue use `{ [key: string]: BodyDataValue }` or changing linting rules.

Reply:
Okay, let me remove it